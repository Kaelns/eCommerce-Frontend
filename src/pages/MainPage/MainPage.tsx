import type { SxPropsObj } from '@/shared/model/types';

import { getErrorMessage } from '@/shared/api/ecommerce-api';

import { MainSection } from '@/pages/MainPage/ui/MainSection';

import { PromocodeSection } from '@/widgets/PromocodeSection';
import { ProductShowcaseSection } from '@/widgets/ProductShowcaseSection';

import { selectLanguage } from '@/entities/user';
import { useGetCategoriesQuery } from '@/entities/categories';

import { SuspenseWithError } from '@/shared/ui/components';
import { useAppSelector } from '@/shared/lib/redux';

const sxContainer: SxPropsObj = {
  display: 'flex',
  flexDirection: 'column',
  gap: 5
};

export function MainPage() {
  const language = useAppSelector(selectLanguage);

  const { data: categoriesCollection, error, isError, isLoading } = useGetCategoriesQuery();

  const firstCategory = categoriesCollection?.categories[0];
  const secondCategory = categoriesCollection?.categories[1];

  return (
    <SuspenseWithError settings={{ isError, isLoading, error: getErrorMessage(error) }} sx={sxContainer}>
      <>
        <MainSection />
        <ProductShowcaseSection categoryId={firstCategory?.id ?? ''} categoryName={firstCategory?.name[language] ?? ''} />
        <ProductShowcaseSection categoryId={secondCategory?.id ?? ''} categoryName={secondCategory?.name[language] ?? ''} />
        <PromocodeSection />
      </>
    </SuspenseWithError>
  );
}
