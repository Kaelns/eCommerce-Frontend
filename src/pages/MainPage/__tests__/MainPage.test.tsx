import { MainPage } from '@/pages/MainPage/MainPage';
import { render } from '@testing-library/react';
import * as useAuthContextHook from '@/context/AuthContext/useAuthContext';

describe('Given MainPage component: ', () => {
  it('when rendered, should match snapshot', () => {
    const useMoviesSpy = vi.spyOn(useAuthContextHook, 'useAuthContext');

    useMoviesSpy.mockReturnValue({
      authTokens: {
        token: '',
        anonToken: ''
      },
      setAuthTokens: vi.fn()
    });

    const { asFragment } = render(<MainPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
