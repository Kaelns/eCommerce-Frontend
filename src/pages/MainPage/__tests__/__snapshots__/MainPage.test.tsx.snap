// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given MainPage component:  > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-ew8gbl-MuiStack-root"
  >
    <section
      class="MuiStack-root css-1u49h8d-MuiStack-root"
    >
      <div
        class="MuiBox-root css-1gesks6"
      />
      <img
        alt="furniture"
        class="MuiBox-root css-h473ni"
        src="/src/assets/furniture.webp"
      />
      <div
        class="MuiStack-root css-14kkdmk-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1y497ks-MuiTypography-root"
        >
          Radiocommerce
        </h1>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1wk7ua4-MuiTypography-root"
        >
          All sort of high-quality furniture available here
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1sr8arn-MuiTypography-root"
        >
          Take a look around and choose what you like.Trust me, you won't regret it. Besides, we have a lot of discounts!
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-16c6ckb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Go to catalog
        </button>
      </div>
    </section>
    <section
      class="MuiStack-root css-1ucnu3w-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-aie56f-MuiPaper-root"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1b1mdw6-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiStack-root css-r78jfo-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-7bduwa-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
          </div>
          <div
            class="MuiStack-root css-x6bnna-MuiStack-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1eogyen-MuiGrid-root"
            />
          </div>
        </div>
      </div>
    </section>
    <section
      class="MuiStack-root css-1ucnu3w-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-aie56f-MuiPaper-root"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1b1mdw6-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiStack-root css-r78jfo-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-7bduwa-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
            />
          </div>
          <div
            class="MuiStack-root css-x6bnna-MuiStack-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1eogyen-MuiGrid-root"
            />
          </div>
        </div>
      </div>
    </section>
    <section
      class="MuiStack-root css-16aaozm-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-kcot8x-MuiPaper-root"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        >
          Our special offer:
        </p>
      </div>
      <div
        class="MuiStack-root css-1ktyeuu-MuiStack-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-sl6uf3-MuiPaper-root"
          style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
          >
            minus10
          </p>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-sl6uf3-MuiPaper-root"
          style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
          >
            minus15
          </p>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
